src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -149,8 +149,15 @@ public async Task InitializeAsync(CancellationToken cancellationToken)
                     SetCurrentCircuitHost(this);
                     _initialized = true; // We're ready to accept incoming JSInterop calls from here on
 
-                    await OnCircuitOpenedAsync(cancellationToken);
-                    await OnConnectionUpAsync(cancellationToken);
+                    if (!existingCircuit)
+                    {
+                        // If the circuit we are using was created during prerendering don't run
+                        // the lifecycle events again at this time.
```

Comment:
Was this a bug before? I'm surprised we need to change that behavior unless it was a bug before.

Reply:
I am a bit uncertain about the meanings of these lifecycle events now. Previously I would have expected:

 * `OnCircuitOpenedAsync` to occur during prerendering
 * `OnConnectionUpAsync` to occur after the client opens the socket connection and is ready for interactivity

... but it seems with this change that there wouldn't be a way to know when the socket connection was opened. Is this right? I'm not sure we've got a formal definition of when these lifecycle events should fire, but now would be a good time to be explicit about it.

Reply:
The way things work now:

Prerendered components or prerendered+non-prerendered
* Prerender -> Runs on Circuit Opened
* Connect -> Runs OnConnectionUp
* StartCircuit -> Renders additional components

Only non prerendered components
* StartCircuit runs OnCircuitOpenedAsyn -> OnConnectionUpAsync -> RendersAdditional components. 

OnConnectionUp always marks when the socket is opened and we're ready to perform JSInterop.